File: 1.Percolation/QuickFind.java
public class QuickFind
{
	public static int[] id;

	public void QuickFindUF(int N)
	{
		id = new int[N];
		for (int i = 0; i < N; i++)
		{
			id[i] = i;
		}
	}
	public boolean connected(int p, int q)
	{ return id[p] == id[q]; }
	public void union(int p, int q)
	{
		int pid = id[p];
		int qid = id[q];
		for (int i = 0; i < id.length; i++)
		{
			if (id[i] == pid) id[i] = qid;
		}	
		
	}
	// Test with an array of length
	public static void main(String[] args) 
	{
		QuickFind my_test = new QuickFind();
		my_test.QuickFindUF(10);
		my_test.union(5, 0);
		my_test.union(1, 9);
		my_test.union(5, 3);
	    my_test.union(8, 3);
		my_test.union(5, 2);
		my_test.union(7, 5);
		// my_test.union(5, 0);
		// my_test.union(7, 2);
		// my_test.union(6, 1);
		for (int i = 0; i < my_test.id.length; i++)
		{
			System.out.print(id[i] + " ");

		}
		System.out.print("\n");



	}
}

File: 1.Percolation/quick_find.py

id_array = range(10)

def union(arr, p, q):
	"""
	conntects 2 components
	"""
	pid = arr[p]
	qid = arr[q]
	for idx in range(len(arr)):
		#print idx
		if arr[idx] == pid:
		#	print idx
			arr[idx] = qid;
	return arr

def connected(arr, p, q):
	"""
	checks if 2 components are connected
	"""
	return arr[p] == arr[q]


print union(id_array, 4, 9)
print union(id_array, 9, 1)
print union(id_array, 0, 9)
print union(id_array, 0, 5)
print union(id_array, 6, 9)
print union(id_array, 6, 7)


#print id_array
